recipe/deploy/update_code.php

```diff
@@ -70,55 +71,68 @@
  */
 desc('Updates code');
 task('deploy:update_code', function () {
-    $git = get('bin/git');
-    $repository = get('repository');
+    $strategy = get('update_code_strategy');
     $target = get('target');
 
-    if (empty($repository)) {
-        throw new ConfigurationException("Missing 'repository' configuration.");
-    }
-
     $targetWithDir = $target;
     if (!empty(get('sub_directory'))) {
         $targetWithDir .= ':{{sub_directory}}';
     }
 
-    $bare = parse('{{deploy_path}}/.dep/repo');
-    $env = [
-        'GIT_TERMINAL_PROMPT' => '0',
-        'GIT_SSH_COMMAND' => get('git_ssh_command'),
-    ];
-
-    start:
-    // Clone the repository to a bare repo.
-    run("[ -d $bare ] || mkdir -p $bare");
-    run("[ -f $bare/HEAD ] || $git clone --mirror $repository $bare 2>&1", env: $env);
-
-    cd($bare);
-
-    // If remote url changed, drop `.dep/repo` and reinstall.
-    if (run("$git config --get remote.origin.url") !== $repository) {
-        cd('{{deploy_path}}');
-        run("rm -rf $bare");
-        goto start;
-    }
-
-    run("$git remote update 2>&1", env: $env);
+    if ($strategy === 'local_archive') {
+        $host = currentHost()->connectionString();
 
+        // Copy to release_path.
+        runLocally(<<<BASH
+            git archive {$targetWithDir} | ssh {$host} "tar -x -f - -C {{release_path}} 2>&1"
```

Comment:
Let's use deployer's upload function.

Reply:
Do you want to use the `upload` function for the tar archive or all the files? I was using `git archive` to respect any `export-ignore`s listed in the `.gitattributes` file.

Are you saying I should:
- locally create the archive
- use the `upload` function to upload the archive
- remotely untar the archive?

Reply:
I've updated